src/pydargs/__init__.py

```diff
@@ -45,11 +45,32 @@ def parse(tp: Type[Dataclass], args: Optional[list[str]] = None, **kwargs: Any)
         An instance of tp.
     """
     namespace = _create_parser(tp, **kwargs).parse_args(args)
-    return tp(**namespace.__dict__)
+    return _create_object(tp, namespace)
+
+
+def _create_object(tp: Type[Dataclass], namespace: Namespace, prefix: str = "") -> Dataclass:
+    for field in fields(tp):
+        if hasattr(field.type, "__dataclass_fields__"):
+            # Create nested dataclass object
+            setattr(
+                namespace,
+                prefix + field.name,
+                _create_object(field.type, namespace, prefix=f"{prefix}{field.name}_"),
+            )
+    args = {key[len(prefix) :]: value for key, value in namespace.__dict__.items() if key.startswith(prefix)}
+    for key in args.keys():  # Remove used keys from the namespace
```

Comment:
```suggestion
    # Remove prefixed keys from the namespace after instantiating nested object.
    for key in args.keys():
```

Reply:
Note that the keys that are removed are not necessarily prefixed, nor for nested objects.

Reply:
Oh dip, it's _that_ generic! Nice. 
However, if possible, I would like to make explicit that this is mostly required for those cases (it is, right? 🤔).
If I understand this correctly: If there are no nested dataclasses, there not going to be a call to this method with a prefix. Meaning this won't recurse and the namespace is cleared but for no reason. So this functionality does purely exist to support that use case...

... and it's not super apparent (to me)

Reply:
How about:

```suggestion
    # Remove the keys used so far from the namespace, to prevent clutter when creating a parent object.
    for key in args.keys():
```

Reply:
:+1: